moveit_core/src/collision_detection/allvalid/collision_env_allvalid.cpp

```diff
@@ -40,7 +40,7 @@
 #include <rclcpp/logging.hpp>
 
 // Logger
-static const rclcpp::Logger LOGGER = rclcpp::get_logger("moveit_collision_detection.collision_world_allvalid");
+static const rclcpp::Logger LOGGER = rclcpp::get_logger("moveit_core.collision_world_allvalid");
```

Comment:
I agree that the logger names should be revised, but setting them all to `moveit_core` doesn't really make them better. Let's omit these changes for now and first discuss how we want to log MoveIt in general.
My thought is that some context is still needed (here maybe `collision_detection`) and that this context should be reflected by file structure and namespace. Ideally we should define a pattern that we could use everywhere, something like `moveit.<package>.<context>.<optional detail>` -> `moveit.core.collision_detection`. (copied to https://github.com/ros-planning/moveit2/discussions/1614#discussioncomment-4390022)

Curiously, this file is `*_env_allvalid` and the logger says `_world_allvalid`...

Reply:
I would suggest we change the logger names to just:

```
<package_name>.<source_file_name>
```

There are two uses of these identifiers I know of:

1. Enabling DEBUG level on a small amount of code to get the debug output.
2. Searching through the code for an ERROR log to figure out where it came from.

I believe the format above would be effective for both of these use cases.

Reply:
third use case is setting up logger filters for debugging of packages/contexts/detail, and with detail I mean plugin or class instance. Actually, adding file would be redundant because that's already reported by the logger.

Reply:
No problem, I can revert these logger name changes. I was even 100% they were worthwhile changes to begin with but I was just doing pattern matching.